### ISSUE TITLE
Curl --cert/-E can't load cert with ':' in filename

### ISSUE THREAD

**Issue by jacobtolar** [curl/curl#9346]
### I did this

```
curl --key /path/to/key --cert /path/to/some-name:not-a-password.cert https://.../
```

### I expected the following
Successful use of client cert

### Observed behavior

```
curl: (58) could not load PEM client certificate, OpenSSL error error:02001002:system library:fopen:No such file or directory, (no key found, wrong pass phrase, or wrong file format?)
```

### curl/libcurl version

Built from master branch just now: 

```
./configure --disable-shared --enable-debug --enable-maintainer-mode LDFLAGS="-L/opt/homebrew/Cellar/openssl@1.1/1.1.1q/lib" CFLAGS="-I/opt/homebrew/Cellar/openssl@1.1/1.1.1q/include" --with-openssl --prefix=/tmp/curlbuild
make -j8 install

/tmp/curlbuild/bin/curl -V
curl 7.85.0-DEV (aarch64-apple-darwin21.5.0) libcurl/7.85.0-DEV OpenSSL/1.1.1q zlib/1.2.11 libidn2/2.3.3 nghttp2/1.48.0
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: alt-svc AsynchDNS Debug HSTS HTTP2 HTTPS-proxy IDN IPv6 Largefile libz NTLM NTLM_WB SSL threadsafe TLS-SRP TrackMemory UnixSockets
```

### operating system

```
uname -a
Darwin V30123H0G5 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64
```

**Comment by jacobtolar** [comment]
The cert parameter is, off course, documented as:

```
       -E, --cert <certificate[:password]>
```

```
              If curl is built against the NSS SSL library then this option can tell curl the nickname of the certificate to use within the NSS database defined by the
              environment variable SSL_DIR (or by default /etc/pki/nssdb). If the NSS PEM PKCS#11 module (libnsspem.so) is available then PEM files may be loaded. If you want
              to use a file from the current directory, please precede it with "./" prefix, in order to avoid confusion with a nickname.  If the nickname contains ":", it
              needs to be preceded by "\" so that it is not recognized as password delimiter.  If the nickname contains "\", it needs to be escaped as "\\" so that it is not
              recognized as an escape character.
```

But it would be helpful if curl had some way to accept filenames containing a colon. 

The error message could also be made more helpful by including the file path that failed the fopen call.

Relevant bit of code: https://github.com/curl/curl/blob/a8f52ce225cc31a5d61f7a8de5da8c7c64834a50/src/tool_getparam.c#L350-L449

One possible solution might be to allow the user to escape the ':' in the parameter to avoid interpreting the portion after the colon as a password. E.g. `--cert /path/to/some\:cert.pem:password`. I didn't check the code (yet) closely enough to verify whether this is possible without breaking some existing behavior.

I am happy to work on a PR with that or another change that you suggest. 

If it's working as designed and there is no desire to change the behavior, I can send a PR to update the docs to call out this caveat in the --cert section.

**Comment by bagder** [comment]
> One possible solution might be to allow the user to escape the ':' in the parameter to avoid interpreting the portion after the colon as a password. E.g. --cert /path/to/some\:cert.pem:password

Except of course that doing with a backslash would be complicated on regular unix shells, as the shell themselves usually use the backslash for escaping.

> If it's working as designed and there is no desire to change the behavior, I can send a PR to update the docs to call out this caveat in the --cert section.

Well, the functionality is known and documented. It has not been much of a problem to users in the past - I guess most people can work around the limitation quite easily. If you think you can make this clearer in the documentation, please give it a shot!

**Comment by jacobtolar** [comment]
Oof, PEBKAC. The escaping I suggested already works fine, I just typo'd something when I was testing it. I'd blame it on today being Monday, but past experience suggests that I'm bound to do something just as silly every other day this week. 😵‍💫

I do think calling out that the escape sequences apply to both regular paths and NSS nicknames would be helpful. The help text currently implies that the escape logic only applies for NSS databases nicknames (`"If the nickname contains ":", it needs to preceded by "\" so that it is not recognized as password delimiter."`). I can send a PR with updated text there.

**Comment by bagder** [comment]
Fixed in #9349
